mkdocs/docs/example-usage/xcm.md

```diff
@@ -0,0 +1,1010 @@
+
+In this guide, we'll show you first how to build the transfer data if you already know the route you want to use and don't need chain or asset information. Then, we'll show you how to build the transfer data if you need to retrieve the list of supported assets and chains for a given asset, which is useful if you're building a UI to allow users to select the asset, source, and destination chains.
+
+
+### Simple Example {: #build-xcm-transfer-data-simple }
+In this example, we want to transfer DOT from Polkadot to Moonbeam. So to get the transfer data, we'll need to set the asset, source, and destination chains. First we'll need to instantiate the SDK, by calling the [`Sdk`](../reference/xcm.md#the-sdk-method) method and then calling the `setAsset`, `setSource`, and `setDestination` methods.
+You can optionally pass in the ecosystem to the `Sdk` method, but in this example, we know the route we want to use, so there is no need to pass in the ecosystem.
+
+```js
+import { Sdk } from '@moonbeam-network/xcm-sdk';
+import { dot, polkadot, moonbeam } from '@moonbeam-network/xcm-config';
+
+const fromPolkadot = async () => {
+  const transferData = await Sdk()
+  .setAsset(dot)
+  .setSource(polkadot)
+  .setDestination(moonbeam)
+.setAddresses({
+  sourceAddress: pair.address,
+    destinationAddress: account.address,
+  });
+};
+
+fromPolkadot();
+```
+	
+### Example with assets and chains information {: #build-xcm-transfer-data-information }
+
+To get started, you'll use the [`Sdk`](../reference/xcm.md#the-sdk-method) method, which eventually will return the transfer data after calling a series of chained methods. In this case you'll want to include the ecosystem, as you'll need to retrieve the list of supported assets and chains for the asset you want to transfer.
+
+
+```js
+import { Sdk } from '@moonbeam-network/xcm-sdk';
+import { Ecosystem } from '../../packages/types/build';
+
+const sdkInstance = Sdk({ ecosystem: Ecosystem.Polkadot });
+```
+
+The chained methods will provide data on the assets and chains along the way, but the final method will return the transfer data. The process of calling the methods is as follows:
+
+1. Get the list of supported assets for the specified ecosystem
+
+    ```js
+    const { assets, setAsset } = sdkInstance;
+    ```
+
+2. Call the `setAasset` function and pass in the key or asset object (which includes the key and the origin symbol) to define the asset to be transferred. For example:
+
+    ```js
+    import { dot } from '@moonbeam-network/xcm-config';
+
+    // Using the key
+    const { sources, setSource } = setAsset(dot);
+    ```
+
+    This will return a list of the supported source chains for this asset and the [`setSource`](../reference/xcm.md#the-sdk-method) function, which is used to define the source chain to transfer the asset from
+
+3. Call the `setSource` function and pass in the chain object (which includes the key, name, and chain type). For example:
```

Comment:
Actually, you can pass a chain or `chainKey` and I think even `{ key: 'polkadot' }` will also work.